R/validate_PsN_options.R

```diff
@@ -1,26 +1,60 @@
 #' Checks valid options for PsN commands
 #'
 #' @param command PsN command.
+#' @param tool PsN tool.
+#' @param installPath Installation path for Perl / PsN
+#' @param version Version of PsN
 #' @param psnOpts List of additional PsN command line options 
 #' (format: optionName = value or optiontName=TRUE/FALSE )
+#' @details Either specify an explicit command to run at the system prompt 
+#' or specify a combination of tool, installPath and version.
 #' @return character string of valid PsN arguments
 #' @examples
-#' validate_PsN_options(tool='execute', psnOpts = list(picky=TRUE, retries=3, 
+#' validate_PsN_options(command='execute', 
+#' psnOpts = list(picky=TRUE, retries=3, 
 #' tweak_inits=TRUE))
 #' @export
 
-validate_PsN_options <- function(command = NULL, psnOpts = NULL, 
-                                 perl_options_command = "psn_options") {
+validate_PsN_options <- function(command = NULL, 
+                                 tool = NULL,
+                                 installPath = NULL,
+                                 version = NULL,
+                                 psnOpts = NULL) {
 
   require(dplyr)
+  require(stringr)
   
-  if (is.null(command)) stop("Expecting a valid PsN command")
+  if(is.null(tool) && is.null(command)) stop("One of tool or command must be specified")
   
-  psnOptionCommand <- defineExecutable(command = "psn_options*")
-  psnCommon <- system( paste(psnOptionCommand, "-h"), intern = TRUE)
+  # If using tool then specify installPath and version
+  if (is.null(command) && !is.null(tool)){
+    if (is.null(installPath)) stop("If using tool, please specify an installPath")
+    if (is.null(version)) stop("If using tool, please specify a version")      
+  }
+  
+  psnOptionCommand <- ifelse( is.null(command), 
+                              defineExecutable(tool = "psn_options", 
+                                               version = version,
+                                               installPath = installPath),
+                              defineExecutable(command = file.path(
+                                dirname(
+                                  stringr::word(command,1)
```

Comment:
OK, so it should be "depends" then rather than "suggests". I'll change that. I'll take a look at the other packages in "suggests" because they're probably in the same category.